apps/youtube_qa/multi_youtube_video_qa/multi_youtube_video_qa.py

```diff
@@ -0,0 +1,324 @@
+        try:
+            transcripts.append(download_youtube_video_transcript(yt_url))
+        except Exception as e:
+            print(e)
+            print(
+                "Failed to download video transcript. Will try downloading video and generating transcript later..."
+            )
+            failed_download_links.append(yt_url)
+            continue
+    
+
+    try:
+        print("establish evadb api cursor")
+        cursor = evadb.connect().cursor()
+
+        grouped_transcript = []
+        print("Create transcript csv")
+        if len(transcripts) > 0:
+            for transcript in transcripts:
+                group_transcript(transcript, grouped_transcript)
+
+            # print(grouped_transcript)
+            df = pd.DataFrame(grouped_transcript)
+            # df = pd.DataFrame(transcripts) 
+            # print(df)
+            if os.path.exists("transcript.csv"):
+                df.to_csv("transcript.csv", mode='a')
+            else:
+                df.to_csv("transcript.csv")
+
+        print("Failed downloads : ", failed_download_links)
+
+        # download youtube video online if the video disabled transcript
+        for yt_url in failed_download_links:
+            print("Downloading : ", yt_url)
+            try:
+                download_youtube_video_from_link(yt_url)
+            except Exception as e:
+                print(f"Downloading {yt_url} failed with ", e)
+                continue
+        
+        # generate video transcript for the downloaded videos
+        current_directory = os.getcwd()
+        files = os.listdir(current_directory)
+        mp4_files = [file for file in files if file.endswith(".mp4")]
+        if not mp4_files:
+            print("No mp4 files found in current directory. Not generating video transcripts .....")
+        else:
+            raw_transcript_string = generate_online_video_transcript(cursor)           
+            partitioned_transcript = partition_transcript(raw_transcript_string)
+            df = pd.DataFrame(partitioned_transcript)
+            print(df)
+            if os.path.exists("transcript.csv"):
+                df.to_csv("transcript.csv", mode='a')
+            else:
+                df.to_csv("transcript.csv")
+
+        print("Total transcription time : ", total_transcription_time)
+
+        load_start_time = time.time()
+        # load chunked transcript into table
+        cursor.query("""DROP TABLE IF EXISTS Transcript;""").execute()
+        cursor.query(
+            """CREATE TABLE IF NOT EXISTS Transcript (text TEXT(100));"""
+        ).execute()
+        cursor.load("transcript.csv", "Transcript", "csv").execute()
+        print(f"Loading transcripts into DB took {time.time() - load_start_time} seconds")
+        
+        print("Creating embeddings and Vector Index")
+        
+        udf_query = """CREATE UDF IF NOT EXISTS embedding
+            IMPL  '../../../evadb/udfs/sentence_feature_extractor.py';
```

Comment:
perhaps use create_udf api instead of the query. Plus, it is perhaps a better practice to declare this path as a macro  so that we can change them more easily. Check youtube_qa.py in master branch.

Reply:
Sure